root/static/scripts/edit/URLCleanup.js

```diff
@@ -5873,6 +5857,47 @@ entitySpecificRules.release_group = function (url) {
   return {result: true};
 };
 
+/**
+ * Allows for every combination of the given types for the given entity type.
+ * Useful for cases where we want to restrict the options to just a specific
+ * subset of relationships (e.g. "get the music" ones), but without making
+ * extra assumptions about how that subset can be used.
+ *
+ * e.g: anyCombinationOf('release', [1, 2, 3]) allows for:
+ * 1, 2, 3, [1, 2], [1, 3], [2, 3], [1, 2, 3]
+ */
+function anyCombinationOf(
+  entityType: RelatableEntityTypeT,
+  types: $ReadOnlyArray<string>,
+): $ReadOnlyArray<EntityTypesMap> {
+  const validCombinations = Array(1 << types.length).fill().map(
+    (e1, i) => types.filter((e2, j) => i & 1 << j),
+  );
+
+  const result = [];
+  for (const combination of validCombinations) {
+    let option;
+    if (combination.length === 0) {
+      // The empty subset is technically valid, but not of interest to us
+      continue;
+    }
+    if (combination.length === 1) {
+      // One-type subsets are expected as a string, not a 1-element array
+      option = combination[0];
+    } else {
+      option = combination;
+    }
+
+    const resultOption: {
+      [entityType: RelatableEntityTypeT]: EntityTypes,
+    } = {};
+    resultOption[entityType] = option;
+    result.push(resultOption);
+  }
+
+  return result;
```

Comment:
It seems to be a pretty complicated implementation but I don’t have suggestion at hand.

Reply:
Happy to get any suggestions to simplify it, but this is what I came up with that worked after fighting it for a while back in the day. Maybe there's a much simpler option though :)

Reply:
You can actually combine things into a single loop, but I don't know if it makes things much clearer; here's the result:

```diff
diff --git a/root/static/scripts/edit/URLCleanup.js b/root/static/scripts/edit/URLCleanup.js
index 49069c188f..e401c48c8f 100644
--- a/root/static/scripts/edit/URLCleanup.js
+++ b/root/static/scripts/edit/URLCleanup.js
@@ -5911,34 +5911,24 @@ entitySpecificRules.release_group = function (url) {
  * 1, 2, 3, [1, 2], [1, 3], [2, 3], [1, 2, 3]
  */
 function anyCombinationOf(
-  entityType: RelatableEntityTypeT,
+  entityType: string,
   types: $ReadOnlyArray<string>,
 ): $ReadOnlyArray<EntityTypesMap> {
-  const validCombinations = Array(1 << types.length).fill().map(
-    (e1, i) => types.filter((e2, j) => i & 1 << j),
-  );
-
   const result = [];
-  for (const combination of validCombinations) {
-    let option;
+  const numCombinations = (1 << types.length) - 1;
+  for (let i = 1; i <= numCombinations; i++) {
+    const combination = types.filter((e2, j) => i & 1 << j);
     if (combination.length === 0) {
       // The empty subset is technically valid, but not of interest to us
       continue;
     }
     if (combination.length === 1) {
       // One-type subsets are expected as a string, not a 1-element array
-      option = combination[0];
+      result.push({[entityType]: combination[0]});
     } else {
-      option = combination;
+      result.push({[entityType]: combination});
     }
-
-    const resultOption: {
-      [entityType: RelatableEntityTypeT]: EntityTypes,
-    } = {};
-    resultOption[entityType] = option;
-    result.push(resultOption);
   }
-
   return result;
 }
```

Reply:
@yvanzo: opinion? ^ I can commit this change if you like it more.

Reply:
Check the timeline, I approved the pull request before @mwiencek posted his suggestion, just didn’t know he was still at this issue, I have no issue with it.